src/litdata/streaming/reader.py

```diff
@@ -305,6 +304,9 @@ def read(self, index: ChunkedIndex) -> Any:
             # track the new chunk index as the latest one
             self._last_chunk_index = index.chunk_index
 
+        if isinstance(self._item_loader, TokensLoader):
```

Comment:
This needs to be under the previous if, so it is called when the entire chunk has been consumed, not on every read call ;)

Reply:
Thanks for pointing that out, @tchaton! I’ll take a closer look and make the necessary adjustments.